projects/plugins/jetpack/extensions/blocks/blogging-prompt/use-prompt-tags.js

```diff
@@ -31,30 +31,59 @@ export function usePromptTags( promptId, tagsAdded, setTagsAdded ) {
 	// Statuses are 'not-started', 'pending', 'fulfilled', or 'rejected'.
 	const promptTagRequestStatus = useRef( 'not-started' );
 
-	// Get information about tags for the edited post.
-	const { postType, postsSupportTags, tags, tagIds, tagsHaveResolved } = useSelect( select => {
-		const { getEditedPostAttribute } = select( 'core/editor' );
-		const { getEntityRecords, getPostType, hasFinishedResolution } = select( 'core' );
-		const _termIds = getEditedPostAttribute( 'tags' );
+	// Split into separate selectors to maintain referential equality
+	const postType = useSelect(
+		select => select( 'core/editor' ).getEditedPostAttribute( 'type' ),
+		[]
+	);
 
-		const query = {
+	const tagIds = useSelect(
+		select => select( 'core/editor' ).getEditedPostAttribute( 'tags' ) || [],
+		[]
+	);
+
+	const postsSupportTags = useSelect(
+		select => select( 'core' ).getPostType( 'post' )?.taxonomies?.includes( 'post_tag' ),
+		[]
+	);
+
+	// Memoize the query object
+	const query = useMemo( () => {
+		if ( tagIds.length === 0 ) {
+			return null;
+		}
+		return {
 			_fields: 'id,name',
 			context: 'view',
-			include: _termIds?.join( ',' ),
+			include: tagIds.join( ',' ),
 			per_page: -1,
 		};
+	}, [ tagIds ] );
 
-		return {
-			postType: getEditedPostAttribute( 'type' ),
-			postsSupportTags: getPostType( 'post' )?.taxonomies.includes( 'post_tag' ),
-			tagIds: _termIds || [],
-			tags: _termIds && _termIds.length ? getEntityRecords( 'taxonomy', 'post_tag', query ) : [],
-			tagsHaveResolved:
-				_termIds && _termIds.length
-					? hasFinishedResolution( 'getEntityRecords', [ 'taxonomy', 'post_tag', query ] )
-					: true,
-		};
-	}, [] );
+	// Get tags with memoized query
+	const tags = useSelect(
+		select => {
+			if ( tagIds.length === 0 ) {
```

Comment:
Could that condition be moved out of the function passed to `useSelect`? Or would that create an issue?

Reply:
So, `useSelect` itself can't be called conditionally. I believe we could miss important store updates when switching between states as well if we were able to use it conditionally.